vortex-array/src/arrow/executor.rs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use arrow_array::ArrayRef as ArrowArrayRef;
+use arrow_schema::DataType;
+use vortex_compute::arrow::IntoArrow;
+use vortex_error::VortexResult;
+use vortex_session::VortexSession;
+
+use crate::arrays::ListVTable;
+use crate::arrays::VarBinVTable;
+
+/// Trait for executing a Vortex array to produce an Arrow array.
+pub trait ArrowArrayExecutor {
```

Comment:
Is the use of "executor" here 100% necessary? I think `ArrowConverter` or `ArrowConvertible` makes more sense to me and then the trait function would just be `convert`. More of a nit, so feel free to ignore.